Vagrantfile

```diff
@@ -329,6 +330,18 @@ ENV['LC_ALL'] = 'en_US.UTF-8'
 Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
   # VirtualBox
   config.vm.provider :virtualbox do |v|
+    unless Vagrant::Util::Platform.windows?
+      if Process.uid == 0
+        machine_id_file=Pathname.new(".vagrant/machines/default/virtualbox/id")
```

Comment:
I tested it now after executing it on Linux. As normal user that file doesn't exist (also when I execute the machine).
On my machine if I try to execute vagrant with sudo I get an error that my user is not allowed to execute it as root.

Reply:
hmm what's the equivalent?

Reply:
I have `.vagrant.d` folder but inside doesn't have any machines folder.

Reply:
that's with a vagrant VM running? or not created? How does it know the ID of the VM? What's the layout/structure?

Reply:
I have that folder just installing it and includes the various VM and plugins for that user but is not helpful to see if the machine is under root or another user.

It isn't more easy to check what kind of user is executing the vagrantfile? https://stackoverflow.com/questions/52747306/how-to-check-for-root-prvileges-from-the-user-in-ruby

Reply:
To be clear, we are _**only**_ interested in exiting when creating a _new_ VM, this does not affect VMs that have already been created this way

Reply:
it should be checking the local `.vagrant` folder not the folder in the homedir, so for a VVV install at `~/vvv-local` it should look inside `~/vvv-local/.vagrant` not `~/.vagrant.d`. I too have a `~/.vagrant.d/` folder on MacOS that looks the same as that

We can't use the presence of `~/.vagrant.d/data/machine-index/index` though as it's a JSON file listing all known machines, unlike the `id`, whereas the `id` file is just the ID of the machine.

So if you have not created a VM for that VVV install then it is expected that the file and folder does not exist.

Reply:
The logic is:

 ```
if {vvvdir}/.vagrant/machines/default/virtualbox/id exists
    the VM exists, don't abort
else
    it doesnt exist therefore no VM exists
```

Reply:
I misconfused thinking that was a .vagrant folder in the homedir instead the VVV folder itself.